lib/libc/Kconfig

```diff
@@ -54,8 +54,8 @@ menu "C Library"
 choice LIBC_IMPLEMENTATION
 	prompt "C Library Implementation"
 	default EXTERNAL_LIBC if NATIVE_BUILD
+	default PICOLIBC
 	default NEWLIB_LIBC if REQUIRES_FULL_LIBC
-	default PICOLIBC if REQUIRES_FULL_LIBC
 	default MINIMAL_LIBC
```

Comment:
dont we end up with both picolibc and minimal libc set as default? or does order here play a role?

Reply:
Kconfig appears to pick the first which matches and which can be enabled -- the goal is to use picolibc where supported otherwise use minimal C library.